### PR TITLE
add unit test for function ValidateLabelSelector and ValidateDeleteOptions

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for function ValidateLabelSelector and ValidateDeleteOptions
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go [hunk 1]
```diff
@@ -92,3 +93,260 @@ func TestValidateLabels(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateLabelSelector(t *testing.T) {
+	successCases := []metav1.LabelSelector{
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go]: Can you add names for the test cases to make them easier to review?

```go
[]struct{
    name string
    selector metav1.LabelSelector
}
```

Then use the golang subtest with the name https://blog.golang.org/subtests to make it easy to recognize what passes and what fails during runs.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go [hunk 2]
```diff
@@ -92,3 +93,260 @@ func TestValidateLabels(t *testing.T) {
...
+				t.Errorf("MatchLabelErrorCases: case[%d]:%v error details do not include %q: %q", i, matchLabelErrorCases[i], matchLabelErrorCases[i].expect, errs[0].Detail)
+			}
+		}
+	}
+
+	opInOrNotInErrMsg := "must be specified when `operator` is 'In' or 'NotIn'"
+	opExistsOrNotExistsErrMsg := "may not be specified when `operator` is 'Exists' or 'DoesNotExist'"
+	opNotValidErrMsg := "not a valid selector operator"
+
+	matchExpressionErrorCases := []struct {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go]: Why are these separate?  No real objection, just curious.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go (hunk 1 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go]: Done

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go (hunk 2 above)
Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation_test.go]: when this unit test is failed, it helps to  which cases this failed.

**Comment by hanxiaoshuai** [comment]
cc @deads2k

**Pull request closed without merging**